tests/mypy_test.py

```diff
@@ -177,17 +221,29 @@ def main():
                 if mod in seen or mod.startswith("."):
                     continue
                 add_files(files, seen, root, name, args, exclude_list)
+                add_configuration(configurations, seen_dist_configs, distribution)
 
         if files:
+            with tempfile.NamedTemporaryFile("w+", delete=False) as temp:
+                temp.write("[mypy]\n")
+
+                for dist_conf in configurations:
+                    temp.write("[mypy-%s]\n" % dist_conf.module_name)
+                    for k, v in dist_conf.values.items():
+                        temp.write("{} = {}\n".format(k, v))
+
+                config_file_name = temp.name
             runs += 1
             flags = [
                 "--python-version", "%d.%d" % (major, minor),
+                "--config-file", config_file_name,
                 "--strict-optional",
                 "--no-site-packages",
                 "--show-traceback",
```

Comment:
We could instead put all of these options in the config file too, because (unless I'm misremembering) options passed at the command line trump those in a config file. But it's also useful to pass them here, so that we have a minimum set of strictness options that all packages have to obey.

My vote is to leave this as is for now, and revisit it if we see a need for less strict options in some particular package.